stable_baselines3/common/vec_env/stacked_observations.py

```diff
@@ -123,6 +123,11 @@ def update(
             if done:
                 if "terminal_observation" in infos[i]:
                     old_terminal = infos[i]["terminal_observation"]
+                    # Make sure that the terminal observation from the environment
+                    # as array_like so it can be concatenated by numpy.
+                    # If it is not (e.g. a single float) it will be put into a list.
+                    if not hasattr(old_terminal, "__len__"):
+                        old_terminal = [old_terminal]
```

Comment:
btw, why a list and not a numpy array?

Reply:
Because the numpy array would still be a scalar value.
```python
old_terminal = np.array(old_terminal)
len(old_terminal)
# TypeError: len() of unsized object
```
So you would either go
```python
old_terminal = np.array([old_terminal])
```
which is basically an extra step, or
```python
old_terminal = np.array(old_terminal).reshape((1,))
```
which I think is unnecessarily unreadable.